h/static/scripts/controllers/lozenge-controller.js

```diff
@@ -1,70 +1,65 @@
-    let lozengeEl = document.createElement('div');
-    let lozengeMarkup = escapeHtml(options.content);
-    const currentLozenges = containerEl.querySelectorAll('.lozenge');
 
-    if (searchTextParser.hasKnownNamedQueryTerm(options.content)) {
-      const queryTerm = searchTextParser.getLozengeFacetNameAndValue(options.content);
-      lozengeMarkup = '<span class="lozenge__facet-name">' +
-        escapeHtml(queryTerm.facetName) +
-        '</span>' +
-        ':' +
-        '<span class="lozenge__facet-value">' +
-        escapeHtml(queryTerm.facetValue) +
-        '</span>';
-    }
-    lozengeEl.innerHTML =
-      '<div class="js-lozenge__content lozenge__content">'+
-      lozengeMarkup +
-      '</div>' +
-      '<div class="js-lozenge__close lozenge__close">' +
-      '<img alt="Delete lozenge" src="/assets/images/icons/lozenge-close.svg">' +
-      '</div>';
-    lozengeEl.classList.add('lozenge');
-    lozengeEl.classList.add('js-lozenge');
+  constructor(element, options) {
+    super(element, options);
 
-    lozTempContainer.appendChild(lozengeEl);
+    let facetName = '';
+    let facetValue = options.content;
 
-    // append the lozenge after the last lozenge child
-    // or as the very first element in the container.
-    // We are doing this over appendChild because we don't want to limit
-    // the ability for the container to also hold other elements like an
-    // input field - allowing them to flow together in the same box model
-    if (currentLozenges && currentLozenges.length > 0) {
-      currentLozenges[currentLozenges.length - 1].insertAdjacentHTML('afterend', lozTempContainer.innerHTML);
-    } else {
-      containerEl.insertAdjacentHTML('afterbegin', lozTempContainer.innerHTML);
+    if (searchTextParser.hasKnownNamedQueryTerm(options.content)) {
+      const queryTerm = searchTextParser.getLozengeFacetNameAndValue(options.content);
+      facetName = queryTerm.facetName;
+      facetValue = queryTerm.facetValue;
     }
 
-    // update reference to point to the actual dom element
-    lozengeEl = containerEl.querySelectorAll('.lozenge')[currentLozenges ? currentLozenges.length : 0];
+    element.classList.add('js-lozenge');
 
-    lozengeEl.querySelector('.js-lozenge__close').addEventListener('mousedown', () => {
-      lozengeEl.remove();
+    this.refs.deleteButton.addEventListener('click', (event) => {
+      event.preventDefault();
       options.deleteCallback();
     });
+
+    this.setState({
+      facetName,
+      facetValue,
+      disabled: false,
+    });
+  }
+
+  update(state) {
+    setElementState(this.element, {disabled: state.disabled});
+    let facetName = state.facetName;
+    if (facetName) {
+      facetName += ':';
+    }
+    this.refs.facetName.textContent = facetName;
+    this.refs.facetValue.textContent = state.facetValue;
+  }
+
+  inputValue() {
+    if (this.state.facetName) {
+      return this.state.facetName + ':' + this.state.facetValue;
```

Comment:
It's a small worry that this `:` is being inserted and appended in multiple places